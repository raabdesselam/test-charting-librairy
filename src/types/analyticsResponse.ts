

export type AnalyticsResponse<T> = {
  from: string;
  to: string;
  generationmix: T[];
};