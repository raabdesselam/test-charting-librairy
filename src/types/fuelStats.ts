export type FuelStat = {
  fuel: string;
  perc: number;
};
