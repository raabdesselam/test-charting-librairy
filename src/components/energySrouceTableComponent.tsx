import React, { FC } from "react";
import { Table, TableProps } from "antd";
import { AnalyticsResponse } from "../types/analyticsResponse";
import { FuelStat } from "../types/fuelStats";

export interface EnergySourceTableComponentProps extends TableProps {
  report: AnalyticsResponse<FuelStat>;
}

export const EnergySourceTableComponent: FC<EnergySourceTableComponentProps> = ({
  report,
  ...props
}) => {
  const columns = [
    {
      title: "Energy Source",
      dataIndex: "energySource",
      key: "energySource",
    },
    {
      title: "Percentage(%)",
      dataIndex: "percentage",
      key: "percentage",
    },
  ];

  const dataSource = report.generationmix.map((entry) => {
    return {
      key: entry.fuel,
      energySource: entry.fuel,
      percentage: entry.perc,
    };
  });

  return <Table dataSource={dataSource} columns={columns} {...props} />;
};
