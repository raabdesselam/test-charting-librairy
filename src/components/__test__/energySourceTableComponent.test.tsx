import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer  from "react-test-renderer";
import "@testing-library/jest-dom";
import { EnergySourceTableComponent } from "../energySrouceTableComponent";
import { AnalyticsResponse } from "../../types/analyticsResponse";
import { FuelStat } from "../../types/fuelStats";

//https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
beforeAll(() => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), 
      removeListener: jest.fn(), 
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
});

afterAll(() => {
  cleanup();
});

const report: AnalyticsResponse<FuelStat> = {
  from: "2019-08-12T12:30Z",
  to: "2019-08-12T13:00Z",
  generationmix: [
    {
      fuel: "biomass",
      perc: 29.8,
    },
    {
      fuel: "coal",
      perc: 70.2,
    },
  ],
};

test("table report should render correctly", () => {
  render(<EnergySourceTableComponent {...{ report: report }} />);
  const table = screen.getByRole("table");

  expect(table).toHaveTextContent("Energy Source");
  expect(table).toHaveTextContent("Percentage");
});


test("table report should match snapshot", () => {
  const tree = renderer.create(<EnergySourceTableComponent {...{ report: report }} />).toJSON();
  expect(tree).toMatchSnapshot()
});
