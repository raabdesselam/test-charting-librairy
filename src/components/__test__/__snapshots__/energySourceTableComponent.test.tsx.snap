// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="ant-table-wrapper css-dev-only-do-not-override-1uweeqc"
  style={{}}
>
  <div
    className="ant-spin-nested-loading css-dev-only-do-not-override-1uweeqc"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table css-dev-only-do-not-override-1uweeqc"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={{}}
          >
            <table
              style={
                {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup />
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    scope="col"
                    style={{}}
                  >
                    Energy Source
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    scope="col"
                    style={{}}
                  >
                    Percentage(%)
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key="biomass"
                  onClick={[Function]}
                  style={{}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={{}}
                  >
                    biomass
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={{}}
                  >
                    29.8
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key="coal"
                  onClick={[Function]}
                  style={{}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={{}}
                  >
                    coal
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={{}}
                  >
                    70.2
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-1uweeqc"
        style={{}}
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
          title="1"
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-next ant-pagination-disabled"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`table report should match snapshot 1`] = `
<div
  className="ant-table-wrapper css-dev-only-do-not-override-1uweeqc"
  style={{}}
>
  <div
    className="ant-spin-nested-loading css-dev-only-do-not-override-1uweeqc"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table css-dev-only-do-not-override-1uweeqc"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={{}}
          >
            <table
              style={
                {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup />
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    scope="col"
                    style={{}}
                  >
                    Energy Source
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    scope="col"
                    style={{}}
                  >
                    Percentage(%)
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key="biomass"
                  onClick={[Function]}
                  style={{}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={{}}
                  >
                    biomass
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={{}}
                  >
                    29.8
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key="coal"
                  onClick={[Function]}
                  style={{}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={{}}
                  >
                    coal
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={{}}
                  >
                    70.2
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-1uweeqc"
        style={{}}
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
          title="1"
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-next ant-pagination-disabled"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
